### ISSUE TITLE
unknown warning option '-Wimplicit-fallthrough=0'

### ISSUE THREAD

**Issue by oclyke** [micropython/micropython#7546]
Just went to build the ```dev``` variant of the unix port and got this error:

```
CC ../../lib/btstack/src/ad_parser.c
error: unknown warning option '-Wimplicit-fallthrough=0'; did you mean '-Wimplicit-fallthrough'? [-Werror,-Wunknown-warning-option]
make: *** [build-incident/lib/btstack/src/ad_parser.o] Error 1
```

Building on: macOS 10.15.7 (19H1217)
From commit: 8be29b9

I modified the file ```extmod/btstack/btstack.mk``` to omit the offending option entirely and was able to proceed through a successful build. No warnings were generated.

I understand this flag was added recently (#6578). Could this be a linux vs. osx issue?

**Comment by dpgeorge** [comment]
This looks like a difference between gcc and clang.  Probably that warning can be omitted when building with clang (eg put in the `ifneq ($(CC),clang)` part of the Makefile).

Might be a good idea to build `micropython-dev` on the mac CI.

**Comment by dpgeorge** [comment]
See #7547 for a fix, and also enabling of `micropython-dev` build on macOS CI.

**Comment by dpgeorge** [comment]
Should be fixed by 74db526cf059d1bc30be38c28c36cf5d1156ae53
